tests/test/transport/test_message_server.cpp

```diff
@@ -163,4 +186,56 @@ TEST_CASE("Test multiple clients talking to one server", "[transport]")
 
     server.stop();
 }
+
+TEST_CASE("Test client timeout on requests to valid server", "[transport]")
```

Comment:
Not sure if this is the right place to put this test. It requires a custom server, but is really testing functionality in the client.